### ISSUE TITLE
[BE] | Help SF transition to Gov Cloud Plus 

### ISSUE THREAD

**Issue by MickinSahni** [department-of-veterans-affairs/va.gov-team#36167]
## High Level User Story/ies
As a Caregiver of a Veteran, I need to apply for VA benefits so I can receive support.

## Context
> ...upcoming migration to Gov Cloud Plus. This migration requires that we set up a test environment, VAGCPlus, to **simulate moving to the new instance that is serviced by a different range of IPs** and **test that all of our current connections and integrations continue to work as expected**. 
>
> With that being said, we would like to request your **assistance in updating our CARMA Online 1010CG connection to point from our Salesforce Staging environment to the new VAGCPlus environment for testing**. Once testing has been completed in VAGCPlus, we will request to switch the connection back to our Staging environment.
>
> 📆 👉🏼 **Testing dates are from 1/27-2/14 and the go live date for the Gov Cloud plus is 2/25-2/26**

## Discovery
- How will we test?
- What role does the CARMA transition to MuleSoft play? (And what is MS team's role?)
- How do we expect the online 10-10CG experience to be impacted? Worst-case scenario?
- What do we (the 10-10 team) need to do? 

### What must be true in order for you to consider this epic complete? 
- Successful tests of VA.gov's Caregiver product against CARMA within Gov Cloud Plus

## Acceptance Criteria
- Successful submissions of the 10-10CG (with different permutations e.g. secondary caregiver)

**Comment by MickinSahni** [comment]
@dillo @jchilders can you brain-dump questions you think we need to answer with this migration? I added a few above.

**Comment by jchilders** [comment]
I just got off a call with Shirley from the Salesforce team about the GC+ migration. Here’s what I learned:
 
Salesforce is currently on Gov Cloud. This effort moves SF from Gov Cloud to Gov Cloud Plus.
There will be new credentials needed to access Salesforce, but this change should otherwise be seamless. IP addresses are changing, hostnames are not. (Shirley is going to confirm this.) All API endpoints and request/response headers will remain the same. I have asked her to send those credentials to myself and Dillo.
ETA for all of this is Feb 25.
 
Now, this could affect our MS migration. Given that we can toggle between MS & SF, we would need to test that both MS and SF/GC+ are working.
 
Honestly, my inclination is to pause MS integration until the GC+ migration has been completed, to be resumed only after we have successfully submitted 10-10CG data to GC+ SF. This would push MS back to early March at the earliest, something I suspect will be met resistance.
 
Pursuing this route is debatable, though, given the expectation that we will be using MS exclusively after it (MS) has been proven out.

**Comment by SKasimow** [comment]
From an email Caitlin sent on 2/4:

```
Jenna reached out to Shirley about the “does VA.gov need a new URL or credentials for production”
question. Please see the response below:
 
For the GovCloud+ migration, there is a new url but only for the testing environment - VACGPlus
It is SSO so you should be able to access it
I see your user is active
```

**Comment by RudyOnRails** [comment]
> From an email Caitlin sent on 2/4:
> 
> ```
> Jenna reached out to Shirley about the “does VA.gov need a new URL or credentials for production”
> question. Please see the response below:
>  
> For the GovCloud+ migration, there is a new url but only for the testing environment - VACGPlus
> It is SSO so you should be able to access it
> I see your user is active
> ```

that makes sense. We were sent new credentials for connecting.

**Comment by RudyOnRails** [comment]
All issues in the epic have been closed. @SKasimow let us know if there's anything else that needs to be done, but closing for now, as this is complete from my perspective.

**Comment by RudyOnRails** [comment]
SF just reported that they moved over their staging.SF env to GovCloudPlus so new credentials are needed for our direct-to-SF connection. Opening this epic again so I can associate a new issue with it. https://github.com/department-of-veterans-affairs/va.gov-team/issues/37929

**Comment by SKasimow** [comment]
@RudyOnRails Can I close this?
